federation/backends/minimal_profile.md

```diff
@@ -0,0 +1,175 @@
+specification, but still be mostly incompatible with other backends, and existing usage patterns. By following this profile, implementors have a better
+view on what is important.
+
+## Scope
+This minimal profile aims to help initial openEO implementations, that seek a least effort approach to achieve minimal compatibility with other backends.
+Requirements in this profile are not sufficient for backends in the openEO platform federation.
+
+## Mandatory endpoints
+
+In the openEO API, all endpoints are optional and discoverable, except for the basic capabilities and versioning endpoints:
+
+https://api.openeo.org/#tag/Capabilities
+
+https://api.openeo.org/#tag/Capabilities/operation/connect
+
+[Authentication](https://api.openeo.org/#section/Authentication) is also mandatory, but this profile does not define a specific method. We highlight that openID connect + PKCE is one of the only standardized and secure methods that are available.
+ 
+
+
+
+Next to those elements, these are the endpoints that are required by this profile:
+
+
+```
+[
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/file_formats"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/collections"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/collections/{collection_id}"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/processes"
+    },
+    {
+      "methods": [
+        "GET",
+        "POST"
+      ],
+      "path": "/jobs"
+    },
+    {
+      "methods": [
+        "GET",
+        "DELETE"
+      ],
+      "path": "/jobs/{job_id}"
+    },
+    {
+      "methods": [
+        "GET",
+        "POST",
+        "DELETE"
+      ],
+      "path": "/jobs/{job_id}/results"
+    }
+  ],
+```
+
+## Batch job support
+
+Support for openEO batch jobs is mandatory, but certain specific elements can be ommitted. The previous section indicates the exact endpoints and HTTP requests that need to be supported.
+
+
+## File formats
+[GeoTiff](./fileformats.md#geotiff) is the only mandatory output format.
+
+## Collections
+A minimal backend shall have at least one raster collection, allowing to build raster data cubes. 
+
+## Processes
+
+### Data Cubes
+
+- `apply`: Apply a process to each pixel
+- `apply_dimension`: Apply a process to pixels along a dimension
+- `dimension_labels`: Get the dimension labels
+- `filter_bands`: Filter the bands by names
+- `filter_bbox`: Spatial filter using a bounding box
+- `filter_spatial`: Spatial filter using geometries
+- `filter_temporal`: Temporal filter for temporal intervals
+- `load_collection`: Load a collection
+- `mask`: Apply a raster mask
+- `resample_spatial`: Resample and warp the spatial dimensions
+- `save_result`: Save processed data
+
+For backends with more than one collection:
+- `merge_cubes`: Merge two data cubes
+- `resample_cube_spatial`: Resample the spatial dimensions to match a target data cube
```

Comment:
Maybe this one is not that minimal. It's from this list the only one we couldn't implemented in our backend. This is certainly due to our approach but maybe also means it's a more complicated to support than others.

Reply:
resample_cube_spatial is now implicitly part of merge_cubes and mask as of v2.0.0 so if we have merge_cubes it seems we should also have resample_cube_spatial. But hearing that SH can't implement it(?), I fear that we may actually regret doing https://github.com/Open-EO/openeo-processes/pull/405 - Thoughts?

Reply:
@dthiex  if you do support merge_cubes, but not resample_cube_spatial, is it possible that cubes on sentinelhub or somehow aligned already by default more easily? Or does merge_cubes result in an error in a lot of cases?

In any case, if you remove it from this list, then there's a lot of example process graphs that don't work on a minimal backend.